contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -507,22 +513,44 @@ tsql_find_coercion_pathway(Oid sourceTypeId, Oid targetTypeId, CoercionContext c
 				ReleaseSysCache(tuple);
 				break;
 			}
+
+			/* We've found VARBINARY To NVARCHAR casting */
+			if (isVarbinary && strcmp(type_nsname, "sys") == 0 && (strcmp(type_name, "nvarchar") == 0))
+				isVarbinaryToNvarchar = true;
+
+			/* Check if type is Varbinary */
+			if (strcmp(type_nsname, "sys") == 0 && strcmp(type_name, "varbinary") == 0)
+				isVarbinary = true;
+
+			/* We've found NVARCHAR TO VARBINARY casting */
+			if (isNvarchar && strcmp(type_nsname, "sys") == 0 && (strcmp(type_name, "varbinary") == 0))
+				isNvarchartoVarbinary = true;
+
+			/* Check if type is NVARCHAR */
+			if (strcmp(type_nsname, "sys") == 0 && strcmp(type_name, "nvarchar") == 0)
+				isNvarchar = true;
+
```

Comment:
Kindly tests casting of User Defined Types on `varchar/nvarchar` to `varbinary/binary`, and vice versa.